lib/pbench/test/unit/common/test_wait_for_uri.py

```diff
@@ -18,19 +18,32 @@ def success(*args: Tuple[str, int], **_kwargs):
         yield None
 
     monkeypatch.setattr(socket, "create_connection", success)
+
     pbench.common.wait_for_uri("http://localhost:42", 142)
     first_arg = called[0][0]
     assert first_arg[0] == "localhost" and first_arg[1] == 42, f"{called[0]!r}"
 
+    pbench.common.wait_for_uri("http://host1.example.com", 143)
+    first_arg = called[0][0]
+    assert first_arg[0] == "host1.example.com" and first_arg[1] == 80, f"{called[0]!r}"
```

Comment:
Wanna try the `user:password@` syntax? It shouldn't matter here since, if your url parse logic works it'll just be dropped; but I wouldn't mind validation that it works.